components/file_system/src/condvar.rs

```diff
@@ -177,7 +177,7 @@ impl Condvar {
         std::mem::drop(guard);
         let timed_out = {
             let f = match node {
-                CondvarNode::Async(ref sem, _) => sem.acquire().fuse(),
+                CondvarNode::Async(ref not, _) => not.notified().fuse(),
```

Comment:
Can we use `notified().await`? Actually, I don't understand what's the purpose of `fuse()` and the `select!`.

Reply:
oh, seems there is no good way to await within a timeout?

Reply:
seems https://docs.rs/tokio/1.0.1/tokio/time/fn.timeout.html timeout would help you to await within a timeout

Reply:
nice